### PR TITLE
Suppress tracebacks.

### PR DESCRIPTION
**Pull request by prescod**
Suppress extra lines in tracebacks as per the comment: # suppress unnecessarily long tracebacks

### CODE REVIEW

**Comment by prescod** [comment]
@agronholm Should I update this? Other than the conflicts, is it the correct implementation?

File: tests/test_typeguard.py [hunk 1]
```diff
@@ -1294,3 +1295,17 @@ def unresolvable_annotation(x: 'OrderedDict'):  # noqa
         assert len(record) == 1
         assert str(record[0].message).startswith("Replaced forward declaration 'OrderedDict' in")
         assert unresolvable_annotation.__annotations__['x'] is collections.OrderedDict
+
+
+class TestTracebacks:
+    def test_short_tracebacks(self):
+        def foo(a: Callable[..., int]):
+            assert check_argument_types()
+        try:
```
Comment (agronholm) [inline tests/test_typeguard.py:1304]: Separate logical blocks with an empty line.

File: typeguard/__init__.py [hunk 2]
```diff
@@ -632,7 +632,7 @@ def check_return_type(retval, memo: Optional[_CallMemo] = None) -> bool:
         try:
             check_type('the return value', retval, memo.type_hints['return'], memo)
         except TypeError as exc:  # suppress unnecessarily long tracebacks
-            raise exc from None
+            raise TypeError(exc) from None
```
Comment (agronholm) [inline typeguard/__init__.py:635]: Use `**exc.args` as argument. Using the previous exception as argument makes the exception look like this: `TypeError(TypeError('argument "a" must be a callable',),)`

File: typeguard/__init__.py [hunk 3]
```diff
@@ -667,7 +667,7 @@ def check_argument_types(memo: Optional[_CallMemo] = None) -> bool:
             try:
                 check_type(description, value, expected_type, memo)
             except TypeError as exc:  # suppress unnecessarily long tracebacks
-                raise exc from None
+                raise TypeError(exc) from None
```
Comment (agronholm) [inline typeguard/__init__.py:670]: As above.

File: typeguard/__init__.py [hunk 4]
```diff
@@ -808,7 +808,10 @@ def wrapper(*args, **kwargs):
         memo = _CallMemo(python_func, _localns, args=args, kwargs=kwargs)
         check_argument_types(memo)
         retval = func(*args, **kwargs)
-        check_return_type(retval, memo)
+        try:
+            check_return_type(retval, memo)
+        except TypeError as exc:
+            raise TypeError(exc) from None
```
Comment (agronholm) [inline typeguard/__init__.py:814]: Ditto.

File: tests/test_typeguard.py [hunk 5]
```diff
@@ -4,6 +4,7 @@
 from concurrent.futures import ThreadPoolExecutor
 from functools import wraps, partial, lru_cache
 from io import StringIO, BytesIO
+import traceback
```
Comment (agronholm) [inline tests/test_typeguard.py:7]: Don't interleave "from" imports and direct imports. Move this to the top.

**Review by agronholm — CHANGES_REQUESTED** [review]
Changes needed – I'll merge once you've made them.

**Review by agronholm — APPROVED** [review]

**Comment by agronholm** [comment]
Thanks!

**Pull request merged**